File: src/main/java/me/tcklpl/naturaldisaster/util/LocationUtils.java
package me.tcklpl.naturaldisaster.util;

import org.bukkit.Location;

public class LocationUtils {

    public static boolean isBetween(Location target, Location a, Location b) {
        var minX = Math.min(a.getX(), b.getX());
        var minY = Math.min(a.getY(), b.getY());
        var minZ = Math.min(a.getZ(), b.getZ());

        var maxX = Math.max(a.getX(), b.getX());
        var maxY = Math.max(a.getY(), b.getY());
        var maxZ = Math.max(a.getZ(), b.getZ());

        return target.getX() >= minX && target.getX() <= maxX &&
                target.getY() >= minY && target.getY() <= maxY &&
                target.getZ() >= minZ && target.getZ() <= maxZ;
    }

}

File: src/main/java/me/tcklpl/naturaldisaster/util/LocationUtils.kt
package me.tcklpl.naturaldisaster.util

import org.bukkit.Location
import kotlin.math.max
import kotlin.math.min

object LocationUtils {

    fun isBetween(target: Location, a: Location, b: Location): Boolean {
        val minX = min(a.x, b.x)
        val minY = min(a.y, b.y)
        val minZ = min(a.z, b.z)

        val maxX = max(a.x, b.x)
        val maxY = max(a.y, b.y)
        val maxZ = max(a.z, b.z)

        return target.x >= minX && target.x <= maxX && target.y >= minY && target.y <= maxY && target.z >= minZ && target.z <= maxZ
    }
}
